iceoryx_utils/include/iceoryx_utils/internal/units/duration.inl

```diff
@@ -31,8 +31,8 @@ inline constexpr Duration::Duration(const Seconds_t seconds, const Nanoseconds_t
         auto additionalSeconds = nanoseconds / NANOSECS_PER_SEC;
         if (std::numeric_limits<Seconds_t>::max() - additionalSeconds < m_seconds)
         {
-            std::clog << __PRETTY_FUNCTION__
-                      << ": Applied values are out of range and would overflow, clamping to max value!" << std::endl;
+            // std::clog << __PRETTY_FUNCTION__
```

Comment:
Oh this is a deep and philosophical. Either we inform the user that something weird is going on but do not have `constexpr` compile time optimization on our hand or we get rid of all those warning messages and have compile time optimization again.

Since we have in every method defined behavior and the behavior is also so that no one should be surprised I tend to get rid of `clog` completely.

@MatthiasKillat what do you think?

Reply:
> I tend to get rid of `clog` completely.

Thumbs up from me!
If we still want to give active feedback we could just let it fail with an assert. This would force the user to handle overflows/negative values and not the duration class.

Reply:
I would add a static_assert here for the debug mode and make in doxygen clear that values are clamped to maximum. The clog can be removed. Same for the other occurences with clog in constexpr functions.

Reply:
@dkroenke sadly you cannot use `static_assert` in this context since you require potential runtime variable knowledge. The best thing would be the usage of `cxx::Expects` and `cxx::Ensures` from `cxx/helplets.hpp`.
They will perform a `std::terminate` if the condition is violated with a detailed error message for the user.

Reply:
@elfenpiff i agree, but i would still let it only terminate in debug mode because clamping a value is imho something which does not need to terminate in release mode

Reply:
@dkroenke I completely agree! But I would still use `cxx::Expects` and `cxx::Ensures` and modify there behavior to the one you suggested. If you are in debug mode you will be terminated here with a detailed error message and in release mode we continue.

But maybe we need to differentiate here a bit. If this is a behavior which is somehow allowed in Release mode and we terminate here only to provide a hint for the developer we maybe should add a parameter to expects and ensures stating that.

```cpp
cxx::Expects( myNumber > 0 ); // hard, calls std::terminate always
cxx::Expects( myNumber > 0, ALGORITHM_CAN_CONTINUE); // with this enum we state that without this 
// requirement fulfilled we can still continue working
```

This would be something for a different pull request but what do you think @dkroenke @elBoberido @marthtz @mossmaurice @budrus

Reply:
I agree with you, modifying `Expects` and `Ensures` depending on release or debug mode is a great solution as trade-off between safety and performance. However, I wouldn't make it more difficult by introducing `ALGORITHM_CAN_CONTINUE`. If utils/hoofs will be used in a broader context, I can predict what will happen: People will always use `ALGORITHM_CAN_CONTINUE`, basically ignoring it and then there will be reports of error in the field because someone forgot to remove it or was aware there even exists another version.. somehow [Hyrum's Law](https://www.hyrumslaw.com/). It's better to keep things simple and stupid.

Reply:
@elfenpiff unfortunately, we can't use `cxx::Expects`/`cxx::Ensures` either. They resolve to `iox::cxx::internal::Require` which is a non-constexpr function.
We also can't use `static_assert` as mentioned before, so I guess we're out of options and for now just remove all `clog`s?

Reply:
Guys, I created #607 to track the suggested changes to `cxx::Expects` and `cxx::Ensures`.
I'll replace all `clog`s now with `/// @todo #607 issue warning or fail`.